cdap-app-fabric/src/main/java/io/cdap/cdap/gateway/handlers/GitHubHttpHandler.java

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright © 2015-2019 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.gateway.handlers;
+
+import com.google.gson.Gson;
+import com.google.inject.Inject;
+import io.cdap.cdap.common.conf.Constants;
+import io.cdap.cdap.gateway.handlers.util.AbstractAppFabricHttpHandler;
+import io.cdap.cdap.internal.app.store.GitHubStore;
+import io.cdap.cdap.internal.github.GitHubRepo;
+import io.cdap.http.HttpResponder;
+import io.netty.handler.codec.http.FullHttpRequest;
+import io.netty.handler.codec.http.HttpRequest;
+import io.netty.handler.codec.http.HttpResponseStatus;
+
+import java.net.HttpURLConnection;
+import java.net.URI;
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
+import java.util.Base64;
+import javax.ws.rs.DELETE;
+import javax.ws.rs.GET;
+import javax.ws.rs.PUT;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+
+@Path(Constants.Gateway.API_VERSION_3)
+public class GitHubHttpHandler extends AbstractAppFabricHttpHandler {
```

Comment:
Add error handling in all methods to return the appropriate HTTP code. For example, if looking for a non existing nickname it should be 404 Not Found, if a duplicate nick name is added it should be 409 Conflict etc.

Reply:
Added error handling for 404 Not Found, 409 Conflict, and 400 Bad Request